src/example-types/results.js

```diff
@@ -24,43 +31,72 @@ module.exports = {
 
     if (context.forceExclude && _.isArray(schema.forceExclude))
       context.exclude = _.union(schema.forceExclude, context.exclude)
+    if (context.include || context.exclude) searchObj._source = {}
+    if (context.include) searchObj._source.includes = context.include
+    if (context.exclude) searchObj._source.excludes = context.exclude
 
-    if (context.include || context.exclude) result._source = {}
-    if (context.include) result._source.includes = context.include
-    if (context.exclude) result._source.excludes = context.exclude
+    // Global schema highlight configuration
+    let schemaHighlight =
+      _.getOr(true, 'highlight', context) &&
+      _.cloneDeep(schema.elasticsearch.highlight)
+    // Specific search highlight override
+    let searchHighlight = _.isPlainObject(context.highlight)
+      ? context.highlight
+      : {}
+    let resultColumns = context.include
 
-    let highlight =
-      _.getOr(true, 'highlight', context) && schema.elasticsearch.highlight
-    let inlineAliases = _.getOr({}, 'inlineAliases', highlight)
-    let showOtherMatches = _.getOr(false, 'showOtherMatches', context)
+    // Highlighting starts with defaults in the schema first
+    if (schemaHighlight) {
+      let showOtherMatches = _.getOr(false, 'showOtherMatches', context)
+      let schemaInline = _.getOr([], 'inline', schemaHighlight)
+      let schemaInlineAliases = _.getOr({}, 'inlineAliases', schemaHighlight)
 
-    if (highlight) {
+      // Concat the search specific override fields with the schema `inline` so we have them as targets for highlight replacement
+      F.setOn(
```

Comment:
Instead of mutating here and having to `_.cloneDeep` above, it's probably better to just use `_.set` and keep it more functionally pure